pkg/types/vsphere/validation/platform.go

```diff
@@ -56,6 +53,227 @@ func ValidatePlatform(p *vsphere.Platform, fldPath *field.Path) field.ErrorList
 		allErrs = append(allErrs, validateDiskType(p, fldPath)...)
 	}
 
+	if len(p.VCenters) > 0 {
+		allErrs = append(allErrs, validateMultiVCenter(p, fldPath)...)
+	}
+
+	// If all VIPs are empty, skip IP validation.  All VIPs are required to be defined together.
+	if strings.Join([]string{p.APIVIP, p.IngressVIP}, "") != "" {
+		allErrs = append(allErrs, validateVIPs(p, fldPath)...)
+	}
+
+	return allErrs
+}
+
+func validateMultiVCenter(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
```

Comment:
The enhancement says:
> Note: Only a single vCenter will
be supported by this effort. While the out-of-tree CCM supports multiple vCenters
the out-of-tree CSI does not.

(sorry, can't find a good way to link the line in the pr)

IIUC it seems like we should add validation to throw an error `if len(p.VCenters) > 1`

Reply:
yeah, good call out there.  i'll add that validation

Reply:
validation added:
https://github.com/openshift/installer/pull/5911/commits/cc00cab911ec2dd6311b87322857ce29ffae6587#diff-2cdb5d09122e1d26502732ec3ee5fb818c7153af73b55c95cc5f41c9ae005febR103-R105
https://github.com/openshift/installer/pull/5911/commits/cc00cab911ec2dd6311b87322857ce29ffae6587#diff-54ccad1462fee98b54d320708c101c5094144e30cf0d9700a44dc8d332898c22R260-R270